### PR TITLE
Add rowStatePreserved property to UIRepeat, exactly the same as UIData

### PR DESCRIPTION
**Pull request by BalusC**
https://github.com/jakartaee/faces/issues/1263

cc: @tandraschko

### CODE REVIEW

File: impl/src/main/java/jakarta/faces/component/UIData.java [hunk 1]
```diff
@@ -16,6 +16,11 @@
 
 package jakarta.faces.component;
 
+import static com.sun.faces.facelets.tag.faces.ComponentSupport.restoreFullDescendantComponentDeltaStates;
```
Comment (mnriem) [inline impl/src/main/java/jakarta/faces/component/UIData.java:19]: This won't work for 5.0 as UIData won't have access to com.sun.faces packages

Comment (mnriem) [reply impl/src/main/java/jakarta/faces/component/UIData.java:19]: I suggest you move the necessary code into a private Utils class (or reuse one of the existing ones if possible) in the jakarta.faces.component package.

**Review by mnriem — CHANGES_REQUESTED** [review]
Make is so the UIData and UIRepeat both use a class from the jakarta.faces.component package (I suggest PackageUtils

**Comment by BalusC** [comment]
Already took into account this while approving your PR :)

**Review by mnriem — APPROVED** [review]

**Comment by BalusC** [comment]
@mnriem there you go: https://github.com/eclipse-ee4j/mojarra/pull/5353

**Pull request merged**